components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationIdentityProviderMgtListener.java

```diff
@@ -172,6 +172,25 @@ public boolean doPostDeleteIdP(String idPName, String tenantDomain) throws Ident
         return super.doPostDeleteIdP(idPName, tenantDomain);
     }
 
+    /**
+     * Clear SP cache after deleting IDPs.
+     *
+     * @param tenantDomain Tenant domain to delete IdPs.
+     * @return
+     * @throws IdentityProviderManagementException
+     */
+    @Override
+    public boolean doPostDeleteIdPsByTenantDomain(String tenantDomain) throws IdentityProviderManagementException {
+
+        // Clear the SP cache since deleted IDP might have contained association with SPs.
+        IdentityServiceProviderCache.getInstance().clear();
```

Comment:
shall we load the SPs of this tenant clear cache separately?

Reply:
Caches will be cleared in the SP deletion

Reply:
Lets remove this